docs/USAGE-GUIDE.md

```diff
@@ -1618,6 +1618,31 @@ Once **s2n_shutdown** is complete:
 * The underlying transport can be closed. Most likely via `close()`.
 * The s2n_connection handle can be freed via [s2n_connection_free](#s2n\_connection\_free) or reused via [s2n_connection_wipe](#s2n\_connection\_wipe)
 
+
+### s2n_mem_set_callbacks
+
+```c
+typedef int (*s2n_mem_init_callback)(void);
+typedef int (*s2n_mem_cleanup_callback)(void);
+typedef int (*s2n_mem_malloc_callback)(void **ptr, uint32_t requested, uint32_t *allocated);
+typedef int (*s2n_mem_free_callback)(void *ptr, uint32_t size);
+
+extern int s2n_mem_set_callbacks(s2n_mem_init_callback mem_init_callback, s2n_mem_cleanup_callback mem_cleanup_callback, s2n_mem_malloc_callback mem_malloc_callback, s2n_mem_free_callback mem_free_callback);
+```
+
+
+**s2n_mem_set_callbacks** allows the caller to over-ride s2n's internal memory
+handling functions. To work correctly, **s2n_mem_set_callbacks** must be called
+before **s2n_init**. **s2n_mem_init_callback** should be a function that will
```

Comment:
You could add a static variable that you set when `s2n_init` is called. Any function that should go before this would error if called after.